### ISSUE TITLE
Custom Mapping as float or integer does not work

### ISSUE THREAD

**Issue by abeltje1** [ankane/searchkick#870]
Dear Searchkick,

I've been trying furiously to change some of my properties to floats and integers via a custom mapping. When I add custom mapping, all goes well. Up until the point where I change my type to float or integer, in that case I get the following error:

`Elasticsearch::Transport::Transport::Errors::BadRequest: [400] {"error":{"root_cause":[{"type":"mapper_parsing_exception","reason":"Mapping definition for [height] has unsupported parameters:  [ignore_above : 256]"}],"type":"mapper_parsing_exception","reason":"Failed to parse mapping [product]: Mapping definition for [height] has unsupported parameters:  [ignore_above : 256]","caused_by":{"type":"mapper_parsing_exception","reason":"Mapping definition for [height] has unsupported parameters:  [ignore_above : 256]"}},"status":400}`

As you can see, the problem seems to lie within the ignore_above parameter. It of course is completely logical that a float or integer should not be accompanied with an ignore_above parameter. Is it possible to keep the searchkick mappings but ditch the ignore above parameter?
When I remove 'merge_mappings:true' it also works, but I'd like to keep the searchkick mappings and only change one property to a float instead of a string. 

Here is my searchkick config for reference: 

```
searchkick  merge_mappings: true, mappings: {
                product: {
                  properties: {
                    height: {type: "float"}
                    }
                }
              }
```

Thanks in advance!

PS awesome gem!

**Comment by abeltje1** [comment]
I have been busy with this and I have found the solution (for me). The core of the problem lies somewhere else entirely!

In my database, I have fields for height, width and depth (it's about products). Those fields are floats in my DB, but were constantly set as strings on my ES instance. I was wondering why. Seems it's because of the 'filterable' option. Height, width and depth were in the filterable array and therefore (for a reason I do not grasp) had to be strings. Your documentation tells me that all fields on which a where query could be applied, should be added to the filterable array. The moment I remove my fields from this array, they become 'normal' floats again in my ES instance and everything works as intended. I can still where-query on height, width and depth even though they are NOT in my filterable array, how is this possible? 

Concluding: I fixed my issue, but I'd like to know firstly why do all filterable fields have to be strings and secondly: why does it seem like filterable doesn't work? I can still run where queries on properties which are not in the filterable array. 

Last but not least: I must say I'm quite noobie when it comes to ES or SearchKick, so excuse me if I say horribly strange things ;)

Cheers,

Abel
